### ISSUE TITLE
Broken on Fedora 29 PHP/Composer - calling composer with not php.ini

### ISSUE THREAD

**Issue by martin-langhoff** [heroku/heroku-buildpack-php#348]
On Fedora 29, using Fedora's PHP and Composer, with heroku buildpack v158 (according to CHANGELOG.md)...

```
$ /usr/local/bin/heroku local 
[OKAY] Loaded ENV .env File as KEY=VALUE Format
11:33:33 AM web.1 |  AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using fe80::cbd0:5dda:5580:ed9f. Set the 'ServerName' directive globally to suppress this message
11:33:33 AM web.1 |  Unable to determine Composer vendor-dir setting; is 'composer' executable on path or 'composer.phar' in current working directory?
[DONE] Killing all processes with signal  SIGINT
11:33:33 AM web.1 Exited with exit code null
```

Exact versions of Fedora components
```
$ rpm -qa php* composer*
...
php-fedora-autoloader-1.0.0-4.fc29.noarch
...
php-cli-7.2.20-1.fc29.x86_64
php-common-7.2.20-1.fc29.x86_64
...
php-fpm-7.2.20-1.fc29.x86_64
php-symfony-finder-2.8.51-1.fc29.noarch

$ rpm -qf /usr/bin/composer
composer-1.8.5-1.fc29.noarch

```

What I suspect happens is that Fedora is using its php.ini to manage the autoloading. So calling `/usr/bin/composer` works, but calling `php -n /usr/bin/composer` breaks. See below:

```
$ /usr/bin/composer config vendor-dir 
vendor
$ php /usr/bin/composer config vendor-dir 
vendor
`$ php -n /usr/bin/composer config vendor-dir 

Fatal error: Uncaught Error: Call to undefined function Composer\Json\json_decode() in /usr/share/php/Composer/Json/JsonFile.php:153
Stack trace:
#0 /usr/share/php/Composer/Factory.php(290): Composer\Json\JsonFile->validateSchema(1)
#1 /usr/share/php/Composer/Factory.php(576): Composer\Factory->createComposer(Object(Composer\IO\ConsoleIO), './composer.json', false)
#2 /usr/share/php/Composer/Console/Application.php(345): Composer\Factory::create(Object(Composer\IO\ConsoleIO), NULL, false)
#3 /usr/share/php/Composer/Console/Application.php(458): Composer\Console\Application->getComposer(false, false)
#4 /usr/share/php/Composer/Console/Application.php(156): Composer\Console\Application->getPluginCommands()
#5 /usr/share/php/Symfony/Component/Console/Application.php(117): Composer\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 /usr/share/php/Composer/Console/Application.php(104): Symfony\Component\Console\Application->run(Object(Symfony in /usr/share/php/Composer/Json/JsonFile.php on line 153
```

**Comment by dzuelke** [comment]
Thanks for this report.

The problem is actually that Fedora (and others) compile all extensions as shared, including e.g. JSON and PCRE, and then load them using individual INIs placed into the `--with-config-file-scan-dir` location.

Using `php -n` prevents all INIs from being loaded, so the extensions don't get loaded because that directory isn't scanned either.

Can you please try branch `php-dash-n-refactor`, see PR #351, and report whether that works for you?

**Comment by dzuelke** [comment]
Alrighty, I have been able to confirm that #351 fixes #345, which has the exact same underlying cause! Merging to master soon; a tagged release will come early next week.

**Comment by dzuelke** [comment]
Merged in 168f61a2f8b8eeb1ff8394227fba170e35851f07
